src/components/NeRadioSelection.vue

```diff
@@ -3,24 +3,26 @@
   SPDX-License-Identifier: GPL-3.0-or-later
 -->
 
-<script lang="ts" setup>
-import { computed, type PropType, type Ref, ref, watch } from 'vue'
-import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
-import { library } from '@fortawesome/fontawesome-svg-core'
-import { faCircleCheck } from '@fortawesome/free-solid-svg-icons'
-import NeFormItemLabel from '@/components/NeFormItemLabel.vue'
-import { v4 as uuidv4 } from 'uuid'
-
-export type RadioCardSize = 'md' | 'lg' | 'xl'
-
-type RadioOption = {
+<script lang="ts">
+export type RadioOption = {
   id: string
   label: string
   description?: string
   icon?: string
   iconStyle?: string
   disabled?: boolean
 }
+</script>
+
+<script generic="T extends RadioOption" lang="ts" setup>
```

Comment:
Why two separate `<script>`s?

Reply:
This is on how the keyword `setup` actually works.
The content of our `script-setup` is actually the content of a bigger function that gets compiled and then served to the mounting point of our application. 
And I cannot define the type `RadioOption` inside the script that is using the same type to create a generic constraint, a reference of this behaviour can be found in the [Vue doc](https://vuejs.org/api/sfc-script-setup.html#usage-alongside-normal-script)